Cork/Views/Sidebar/Components/Formulae Section.swift

```diff
@@ -9,183 +9,56 @@ import SwiftUI
-                                        brewData: brewData,
-                                        appState: appState,
-                                        outdatedPackageTracker: outdatedPackageTracker,
-                                        shouldRemoveAllAssociatedFiles: false,
-                                        shouldApplyUninstallSpinnerToRelevantItemInSidebar: true
-                                    )
-                                }
-                            } label: {
-                                Text("sidebar.section.installed-formulae.contextmenu.uninstall-\(formula.name)")
-                            }
-                            
-                            if allowMoreCompleteUninstallations
-                            {
-                                Button
-                                {
-                                    Task
-                                    {
-                                        try await uninstallSelectedPackage(
-                                            package: formula,
-                                            brewData: brewData,
-                                            appState: appState,
-                                            outdatedPackageTracker: outdatedPackageTracker,
-                                            shouldRemoveAllAssociatedFiles: true,
-                                            shouldApplyUninstallSpinnerToRelevantItemInSidebar: true
-                                        )
-                                    }
-                                } label: {
-                                    Text("sidebar.section.installed-formulae.contextmenu.uninstall-deep-\(formula.name)")
-                                }
-                            }
-                        }
-                    }
-                }
+                ProgressView()
             }
             else
             {
-                ProgressView()
+                ForEach(displayedFormulae)
+                { formula in
+                    SidebarPackageRow(package: formula)
+                }
             }
         }
         .collapsible(true)
     }
+
+    private var displayedFormulae: [BrewPackage]
+    {
+        let filter: (BrewPackage) -> Bool
+
+        if currentTokens.contains(.intentionallyInstalledPackage)
+        {
+            if searchText.isEmpty
+            {
+                filter = \.installedIntentionally
+            } else
+            {
+                filter = { $0.installedIntentionally && $0.name.contains(searchText)  }
+            }
+        } 
+        else
+        {
+            if searchText.isEmpty || searchText.contains("#")
+            {
+                filter = { _ in true }
+            } 
+            else
+            {
+                filter = { $0.name.contains(searchText) }
+            }
+        }
+
+        return brewData.installedFormulae.filter(filter)
+    }
```

Comment:
I felt like this computed property looked a fair bit more readable before applying the project's preferred formatting to be honest 😅